main.go

```diff
@@ -64,68 +47,36 @@ func applyVirtualMachineExport(vmNamespace, vmName string) error {
 		},
 	}
 
-	_, err = client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
+	_, err := client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
 	return err
 }
 
-func downloadVirtualMachineDiskImage(vmName, volumeName string) error {
-	log.Printf("Downloading disk image from %s Virtual Machine...\n", vmName)
-
-	cmd := exec.Command("usr/bin/virtctl", "vmexport", "download", vmName, "--vm", vmName, "--volume", volumeName, "--output", diskPath)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-
-	if err := cmd.Run(); err != nil {
-		return err
-	}
-
-	if fileInfo, err := os.Stat(diskPath); err != nil || fileInfo.Size() == 0 {
-		return fmt.Errorf("File does not exist or is empty.")
-	}
-
-	log.Println("Download completed successfully.")
-	return nil
-}
-
-func decompressVirtualMachineDiskImage() error {
-	log.Println("Decompressing downloaded disk image...")
+func getRawDiskUrlFromVirtualMachineExport(client kubecli.KubevirtClient, vmNamespace, vmName string) (string, error) {
+	log.Println("Getting raw disk URL from VirtualMachineExport object...")
 
-	diskImg, err := os.Open(diskPath)
-	if err != nil {
-		return err
+	vmExport, err := client.VirtualMachineExport(vmNamespace).Get(context.Background(), vmName, metav1.GetOptions{})
+	if vmExport.Status.Links == nil {
+		return "", err
 	}
-	defer diskImg.Close()
 
-	gzipReader, err := gzip.NewReader(diskImg)
-	if err != nil {
-		return err
+	if vmExport.Status.Links.Internal != nil {
+		return "", err
 	}
-	defer gzipReader.Close()
 
-	newDiskImg, err := os.Create(diskPathDecompressed)
-	if err != nil {
-		return err
-	}
-	defer newDiskImg.Close()
-
-	_, err = io.Copy(newDiskImg, gzipReader)
-	if err != nil {
-		return err
-	}
-
-	err = os.Chmod(diskPathDecompressed, 0666) // Grants read and write permission to everyone.
-	if err != nil {
-		return err
+	for _, volume := range vmExport.Status.Links.Internal.Volumes {
+		for _, format := range volume.Formats {
+			if format.Format == v1beta1.KubeVirtRaw {
+				return format.Url, nil
+			}
+		}
 	}
-
-	log.Println("Decompression completed successfully.")
-	return nil
+	return "", nil
 }
 
-func convertRawDiskImageToQcow2() error {
+func convertRawDiskImageToQcow2(rawDiskUrl string) error {
 	log.Println("Converting raw disk image to qcow2 format...")
 
-	cmd := exec.Command("qemu-img", "convert", "-f", "raw", "-O", "qcow2", diskPathDecompressed, diskPathConverted)
+	cmd := exec.Command("qemu-img", "convert", "-t", "none", "-p", "-O", "qcow2", rawDiskUrl, diskPathConverted)
```

Comment:
Implement retry mechanism in case qemu-img failed because of timeout or connection lost (happens quite often).

Reply:
What means the `-t none`?

Reply:
I think it means no cache (from `--help` you can see `[-t cache]`) when writing the disk to the storage.